crypto/src/e2e_identity/id.rs

```diff
@@ -0,0 +1,142 @@
+use crate::{prelude::ClientId, CryptoError, CryptoResult};
+
+#[cfg(test)]
+const DOMAIN: &str = "wire.com";
+const COLON: u8 = 58;
+
+/// This format: 'bd4c7053-1c5a-4020-9559-cd7bf7961954:4959bc6ab12f2846@wire.com'
+#[derive(Debug, Clone, PartialEq, Eq, Hash, derive_more::From, derive_more::Into, derive_more::Deref)]
+pub struct WireQualifiedClientId(ClientId);
+
+#[cfg(test)]
+impl WireQualifiedClientId {
+    pub fn get_user_id(&self) -> String {
+        let mut split = self.0.split(|b| b == &COLON);
+        let user_id = split.next().unwrap();
+        uuid::Uuid::try_parse_ascii(user_id).unwrap().to_string()
+    }
+
+    pub fn generate() -> Self {
+        let user_id = uuid::Uuid::new_v4().to_string();
+        let device_id = rand::random::<u64>();
+        let client_id = format!("{user_id}:{device_id:x}@{DOMAIN}");
+        Self(client_id.into_bytes().into())
+    }
+
+    pub fn to_static_str(&self) -> &'static str {
+        Box::leak(Box::new(self.to_string()))
```

Comment:
Even if this is just test stuff, those intentional memory leaks might bite us later. Maybe finding a better way to do that (just have owned strings cloned around?)